crates/ruff_linter/src/rules/refurb/rules/slice_to_remove_prefix_or_suffix.rs

```diff
@@ -0,0 +1,498 @@
+        return;
+    }
+
+    if let Some(removal_data) = affix_removal_data_expr(if_expr) {
+        if affix_matches_slice_bound(&removal_data) {
+            let kind = removal_data.affix_query.kind;
+            let text = removal_data.text;
+
+            let mut diagnostic = Diagnostic::new(
+                SliceToRemovePrefixOrSuffix {
+                    affix_kind: kind,
+                    string: text.to_string(),
+                    stmt_or_expression: StmtOrExpr::Expression,
+                },
+                if_expr.range,
+            );
+            let replacement =
+                generate_removeaffix_expr(text, &removal_data.affix_query, checker.generator());
+
+            diagnostic.set_fix(Fix::safe_edit(Edit::replacement(
+                replacement,
+                if_expr.start(),
+                if_expr.end(),
+            )));
+            checker.diagnostics.push(diagnostic);
+        }
+    }
+}
+
+/// FURB188
+pub(crate) fn slice_to_remove_affix_stmt(checker: &mut Checker, if_stmt: &ast::StmtIf) {
+    if checker.settings.target_version < PythonVersion::Py39 {
+        return;
+    }
+    if let Some(removal_data) = affix_removal_data_stmt(if_stmt) {
+        if affix_matches_slice_bound(&removal_data) {
+            let kind = removal_data.affix_query.kind;
+            let text = removal_data.text;
+
+            let mut diagnostic = Diagnostic::new(
+                SliceToRemovePrefixOrSuffix {
+                    affix_kind: kind,
+                    string: text.to_string(),
+                    stmt_or_expression: StmtOrExpr::Statement,
+                },
+                if_stmt.range,
+            );
+
+            let replacement = generate_assignment_with_removeaffix(
+                text,
+                &removal_data.affix_query,
+                checker.generator(),
+            );
+
+            diagnostic.set_fix(Fix::safe_edit(Edit::replacement(
+                replacement,
+                if_stmt.start(),
+                if_stmt.end(),
+            )));
+            checker.diagnostics.push(diagnostic);
+        }
+    }
+}
+
+/// Given an expression of the form:
+///
+/// ```python
+/// text[slice] if text.func(affix) else text
+/// ```
+///
+/// where `func` is either `startswith` or `endswith`,
+/// this function collects `text`,`func`, `affix`, and the non-null
+/// bound of the slice. Otherwise, returns `None`.
+fn affix_removal_data_expr(if_expr: &ast::ExprIf) -> Option<RemoveAffixData> {
+    let ast::ExprIf {
+        test,
+        body,
+        orelse,
+        range: _,
+    } = if_expr;
+
+    let ast::ExprSubscript { value, slice, .. } = body.as_subscript_expr()?;
+    let else_or_target_name = &orelse.as_name_expr()?.id;
+    // Variable names correspond to:
+    // ```python
+    // value[slice] if test else else_or_target_name
+    // ```
+    affix_removal_data_core_logic(value, test, else_or_target_name, slice)
```

Comment:
Nit: We normally use `_impl` to signal that this is the main implementation. 

Or you use `affix_remove_data` to indicate that it applies to all nodes.

Reply:
Changed to `affix_removal_data`